### PR TITLE
Delete consult

### LINKED ISSUE #169: Create delete command for Consultations
Delete command needed for consultations

### PR DESCRIPTION
**Pull request by kwannoel**
Add delete consultation command

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/DeleteConsultCommand.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+     * @param targetIndex Index of consultation being deleted.
+     */
+    public DeleteConsultCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Consultation> lastShownList = model.getFilteredConsultList();
```
Comment (Rishi5154) [inline src/main/java/seedu/address/logic/commands/DeleteConsultCommand.java]: Collection naming terminology can be changed to match plural form as per coding standards

File: src/test/java/seedu/address/logic/commands/DeleteConsultCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+
+/**
+ * Contains integration tests (interaction with the Model, UndoCommand and RedoCommand) and unit tests for
+ * {@code DeleteCommand}.
```
Comment (Rishi5154) [inline src/test/java/seedu/address/logic/commands/DeleteConsultCommandTest.java]: `{@code DeleteConsultCommand}` right

File: src/main/java/seedu/address/logic/commands/DeleteConsultCommand.java (hunk 1 above)
Comment (chrisgzf) [inline src/main/java/seedu/address/logic/commands/DeleteConsultCommand.java]: Agree with Rishi,

Can consider:
```suggestion
        List<Consultation> lastShownConsultations = model.getFilteredConsultList();
```

File: src/test/java/seedu/address/logic/commands/DeleteConsultCommandTest.java (hunk 2 above)
Comment (chrisgzf) [inline src/test/java/seedu/address/logic/commands/DeleteConsultCommandTest.java]:
```suggestion
 * Contains integration tests (interaction with the {@code Model}, {@code UndoCommand}
 * and {@code RedoCommand}) and unit tests for {@code DeleteConsultCommand}.
```

**Review by chrisgzf — APPROVED** [review]
Few nits else LGTM!

**Pull request merged**